doc/security/security-best-practice-en.md

```diff
@@ -0,0 +1,83 @@
+# Fate security best practice guidance
+
+When the FATE is used to perform tasks such as federated learning, security models and security protocols should be used to ensure that no privacy data is disclosed during the calculation process. This draft attempts to provide security instructions for using FATE.
+## 1. State of Art of FATE
+
+FATE includes multiple federal learning protocols, some of which are composed of participants and third-party arbiters, such as Homo LR. Other federated learning protocols do not require third-party arbiter to do parameter aggregation, such as secureboost, Hetero Neural Network, Hetero FTL, etc. Therefore, FATE security model can be divided into two types: with third-party arbiter and without third-party arbiter.
+
+For the federated learning without third-party arbiter, its security assumption is: each participant in FATE is an honest, curious and independent security domain, and the data interaction between participants can ensure that the privacy data of each security domain is not leaked. For the federated learning with third-party arbiter, the security assumption is: the federal learning participants are honest, curious and independent security domains, and the arbiter is honest and curious for all participants. That is, the arbiter will comply with the FATE protocol's execution, but may retain or inference aggregation stage intermediate data. When multiple arbiters perform aggregation and arbitration operations, they must not collude with each other.
+
+Security models of FATE need to use security protocols such as secure multi-party computing to protect data security during data interaction stage between different participants. FATE security protocols include Paillier Encryption, RSA Encryption, Hash Factory, DH Key Exchange, SPDZ, OT, etc.
+
+## 2. Network Security Suggestions
+
+For the participating nodes in FATE, client_authentication and permission components need to be used to realize identity authentication function between nodes.
+
+In the network transmission process involved in FATE, such as HTTPS, Flask, gRPC, etc., SSL/TLS protocol is used to protect the security of data transmission. In particular, SSL/TLS is required for network transmission in a public network environment. TLS key strength complies with RFC8446 section9.1.
+
+FATE usually expose one Network port: 9370，network data package of other ports need to be treated with caution.
```

Comment:
Seems this is for eggroll as the federation engine. If we use Spark, then it would be "9300 or 9310" for nginx, "6650 & 6651" for pulsar and "5672" for rabbitmq.